### ISSUE TITLE
Cannot fix "The background services are not running" after upgrade to 15.2.0

### ISSUE THREAD

**Issue by xwillq** [pqrs-org/Karabiner-Elements#3973]
## Environment
macOS version: 15.0
Karabiner version: 15.0.0, 15.1.0, 15.2.0, 15.2.1

## Problem
After updating macos to 15.0 and Karabiner to 15.0.0 everything was working fine, but after updating Karabiner to 15.2.0 the following error occurred:
![CleanShot 2024-10-16 at 14 19 39@2x](https://github.com/user-attachments/assets/9a9dd89e-db1f-425c-90c9-8a2c5a229373)

Both background services are enabled:
![CleanShot 2024-10-16 at 14 21 23@2x](https://github.com/user-attachments/assets/2350e87e-0b3a-44e4-a659-3e997f280eed)

To fix this, I have tried:
1. Disabling and enabling background services.
2. Disabling background services, rebooting and enabling them again.
3. Uninstalling Karabiner from Uninstall menu, deleting app files from `~/Library`, `/Library`, `~/.local/share/karabiner`, deleting driver from Driver extensions and installing it again.
4. Upgrading to 15.2.1.
5. Downgrading to other 15.x versions (after uninstalling like in point 3).
6. Downgrading to latest 14.x (after uninstalling like in point 3). It didn't have this problem, but remaps weren't working.
7. Upgrading from latest 14.x to 15.2.0, as if I just updated macos now.
8. Starting Karabiner 15.2.0 in safe boot.
9. Reseting background services using `sfltool resetbtm`.

None of these attempts fixed it for me, `Karabiner-Elements Non-Privileged Agents` still cannot start.

**Comment by Dream-zhc** [comment]
Exactly the same issue here :/

**Comment by NiklasBr** [comment]
Same issue on Mac OS Sonoma 14.7 (23H124) with karabiner 15.2.0

**Comment by Dream-zhc** [comment]
Exactly the same issue here :/

**Comment by mtraton** [comment]
Same here.

**Comment by jabelic** [comment]
Same here on macOS 13.5.1 with karabiner 15.2.0, but it was resolved by UPDATE to 15.2.2.

**Comment by NiklasBr** [comment]
Where do you find 15.2.2? Latest I see on https://github.com/pqrs-org/Karabiner-Elements/releases is 15.2.0

**Comment by jabelic** [comment]
> Where do you find 15.2.2? Latest I see on https://github.com/pqrs-org/Karabiner-Elements/releases is 15.2.0

![CleanShot 2024-10-25 at 15 42 35](https://github.com/user-attachments/assets/5106a6c0-6521-4bc9-8c89-9ca9dfc94a41)

I just followed karabiner-elements > Update > "Check for updates" to check for updates so I updated.
My Karabiner-elements is showing 15.2.2, and it probably means beta.

**Comment by NiklasBr** [comment]
Could be a beta

![Screenshot 2024-10-25 at 09 11 43](https://github.com/user-attachments/assets/8de5ad4a-f598-4b2a-bd27-4e721a7dcb79)

**Comment by xwillq** [comment]
I have installed karabiner 15.3.0 after uninstalling previous version and it fixed the problem for me. Simple update didn't work, so I guess you have to do clean reinstall.

**Comment by ceyhanbekiroglu** [comment]
Updated to beta 15.3.0 and it solved the issue.

**Comment by erickrf** [comment]
I was having the same issue with 15.3.0. It only worked when I enabled permission for Input Monitoring for karabiner_grabber.

**Comment by ltpquang** [comment]
I'm on 15.3.0 and got the same issue. karabiner_grabber already got Input Monitoring permission. Also found out that there are many permission error logs complaining about grabber_client permission. Don't know where to allow it.

```
[2024-11-20 14:23:07.674] [error] [console_user_server] grabber_client connect_failed: Permission denied
```

**Comment by arekdist** [comment]
I'm fighting with this since Sequoia release. Nothing works properly. Grabber has permission's issue. It should be possible to push MDM profile with following:
```
IdentifierType = Path
Identifier = /Library/Application Support/org.pqrs/Karabiner-Elements/bin/karabiner_grabber
Code Requirements = identifier "karabiner_observer" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = G43BCU2T37
```
Profile is distributed but it doesn't work as expected - user is asked for admin's permission still...

**Comment by larrasket** [comment]
Uninstalling Karabiner (w/ brew) and reinstalling it fixed it for me.

**Comment by dws2020** [comment]
Reinstalling (uninstall + install) fixed the issue for me too.
My custom rules/settings live in `~/.config/karabiner`, so no need to reset the config.

macOS Tahoe 26.3 / Karabiner-Elements 15.9.0

**Comment by larrasket** [comment]
Although uninstalling and installing worked, something misses it up later

**Comment by ttscoff** [comment]
Just got my MBP back from the shop. They wiped it and I had to restore. Karabiner gave me the message "The background services are not running." I tried disabling and renabling in System Settings, then tried a full uninstall (via Karabiner settings) and reinstall. Cannot get the Non-privileged daemons marker to switch on.

![](https://cleanshot.com/share/yHPkhfwn+)

Karabiner Version 15.9.0 (15.9.0)
macOS Tahoe 26.5
